### PR TITLE
fix: createNewRecipeUser input in consume code API

### PR DESCRIPTION
**Pull request by sattvikc**
## Summary of change

(A few sentences about this PR)

## Related issues

- https://github.com/supertokens/supertokens-core/issues/554
- https://github.com/supertokens/supertokens-core/pull/651

## Test Plan

(Write your test plan here. If you changed any code, please provide us with clear instructions on how you verified your
changes work. Bonus points for screenshots and videos!)

## Documentation changes

(If relevant, please create a PR in our [docs repo](https://github.com/supertokens/docs), or create a checklist here
highlighting the necessary changes)

## Checklist for important updates

- [ ] Changelog has been updated
    - [ ] If there are any db schema changes, mention those changes clearly
- [ ] `coreDriverInterfaceSupported.json` file has been updated (if needed)
- [ ] `pluginInterfaceSupported.json` file has been updated (if needed)
- [ ] Changes to the version if needed
    - In `build.gradle`
- [ ] If added a new paid feature, edit the `getPaidFeatureStats` function in FeatureFlag.java file
- [ ] Had installed and ran the pre-commit hook
- [ ] If there are new dependencies that have been added in `build.gradle`, please make sure to add them
  in `implementationDependencies.json`.
- [ ] Update function `getValidFields` in `io/supertokens/config/CoreConfig.java` if new aliases were added for any core config (similar to the `access_token_signing_key_update_interval` config alias).
- [ ] Issue this PR against the latest non released version branch.
    - To know which one it is, run find the latest released tag (`git tag`) in the format `vX.Y.Z`, and then find the
      latest branch (`git branch --all`) whose `X.Y` is greater than the latest released tag.
    - If no such branch exists, then create one from the latest released branch.
- [ ] If added a foreign key constraint on `app_id_to_user_id` table, make sure to delete from this table when deleting the user as well if `deleteUserIdMappingToo` is false.
## Remaining TODOs for this PR

- [ ] Item1
- [ ] Item2

### CODE REVIEW

File: src/main/java/io/supertokens/webserver/api/passwordless/ConsumeCodeAPI.java [hunk 1]
```diff
@@ -81,36 +82,46 @@ protected void doPost(HttpServletRequest req, HttpServletResponse resp) throws I
...
+                result.addProperty("createdNewUser", consumeCodeResponse.createdNewUser);
+                result.add("user", userJson);
+                if (getVersionFromRequest(req).greaterThanOrEqualTo(SemVer.v4_0)) {
+                    for (LoginMethod loginMethod : consumeCodeResponse.user.loginMethods) {
+                        if (loginMethod.recipeId.equals(RECIPE_ID.PASSWORDLESS)
+                                && (consumeCodeResponse.email == null || Objects.equals(loginMethod.email, consumeCodeResponse.email))
+                                && (consumeCodeResponse.phoneNumber == null || Objects.equals(loginMethod.phoneNumber, consumeCodeResponse.phoneNumber))) {
+                            result.addProperty("recipeUserId", loginMethod.getSupertokensOrExternalUserId());
+                            break;
+                        }
```
Comment (rishabhpoddar) [inline src/main/java/io/supertokens/webserver/api/passwordless/ConsumeCodeAPI.java:124]: - we also want to return the device
- remove the factorId stuff below + check that it's not there anywhere else too

File: src/test/java/io/supertokens/test/passwordless/api/PasswordlessConsumeCodeAPITest5_0.java [hunk 2]
```diff
@@ -0,0 +1,812 @@
...
+            return;
+        }
+
+        String email = "test@example.com";
+        CreateCodeResponse createResp = Passwordless.createCode(process.getProcess(), email, null, null, null);
+
+        JsonObject consumeCodeRequestBody = new JsonObject();
+        consumeCodeRequestBody.addProperty("deviceId", createResp.deviceId);
+        consumeCodeRequestBody.addProperty("preAuthSessionId", createResp.deviceIdHash);
+        consumeCodeRequestBody.addProperty("userInputCode", createResp.userInputCode + "nope");
```
Comment (rishabhpoddar) [inline src/test/java/io/supertokens/test/passwordless/api/PasswordlessConsumeCodeAPITest5_0.java:446]: - set the new boolean to false here
- add another test that the dfefault value of this boolean is true

**Review by rishabhpoddar — CHANGES_REQUESTED** [review]

File: src/main/java/io/supertokens/webserver/api/passwordless/ConsumeCodeAPI.java (hunk 1 above)
Comment (sattvikc) [reply src/main/java/io/supertokens/webserver/api/passwordless/ConsumeCodeAPI.java:124]: Done

File: src/test/java/io/supertokens/test/passwordless/api/PasswordlessConsumeCodeAPITest5_0.java (hunk 2 above)
Comment (sattvikc) [reply src/test/java/io/supertokens/test/passwordless/api/PasswordlessConsumeCodeAPITest5_0.java:446]: Done

**Review by rishabhpoddar — APPROVED** [review]

**Pull request merged**